src/nvim/msgpack_rpc/channel.c

```diff
@@ -823,6 +823,7 @@ static void log_server_msg(uint64_t channel_id,
   msgpack_unpack_next(&unpacked, packed->data, packed->size, NULL);
   uint64_t type = unpacked.data.via.array.ptr[0].via.u64;
   DLOGN("[msgpack-rpc] nvim -> client(%" PRIu64 ") ", channel_id);
+  log_lock();
```

Comment:
>  the usage of log.c mutex is really simple and won't spread beyond this and the standard log function

It's simple in the context of this PR where we have the state in our heads--but complexity emerges slowly over time. 

I hope we can keep the number of modules that manage synchronization to very few, isolated cases so that it is easy to keep a mental model of the system. Keep in mind that I don't know your plans and I haven't had a chance to review the previous PR, so I can only comment from a "principles" perspective.